### ISSUE TITLE
Show loading message below response loading spinner

### ISSUE THREAD

**Issue by moziliar** [TEAMMATES/teammates#10986]
From PR #10960 

**Description of feature/enhancement**

If a course has many students (or in general many responses per question), add extra loading text below the spinner to inform the user that the students are being loaded (after perhaps 5s).

![image](https://user-images.githubusercontent.com/19870898/109378773-25782300-7910-11eb-82af-dda66fc57782.png)

**Justification**

From @damithc 
> Provided the load operation will not time out, perhaps simply show a message below the spinner cautioning the user that this particular question will take a long time to load?

This would give user more assurance that there is progress in the loading.

**Existing similar features and their shortcomings**

Experimented with using `forkJoin` in the frontend to fetch questions in parallel in #10960: the parallel calls create too much instantaneous load at the backend and causes new instances to be spun up. The cost here is not justified.

**Comment by anshuman-dev** [comment]
Hi. Is this issue resolved? Or anyone working still on it?

**Comment by t-cheepeng** [comment]
@anshuman-dev Priority will be given to the previous commenter as they have a PR up. You can try to find another `good first issue` to work on

**Comment by P-Fabian** [comment]
Is this issue resolved?

**Comment by shivamsharma01** [comment]
Is the issue resolved? I would like to take this up.

**Comment by ypinhsuan** [comment]
@shivamsharma01 The issue is still open, you can work on it

**Comment by hhdqirui** [comment]
@shivamsharma01 Hi, for the first error it may be because your folder `Open Source` contains a space in the folder name? Maybe can try changing the folder name to `OpenSource` or something that do not contain a space.

**Comment by arnavk1872** [comment]
Is this issue still open? I would like to work on it.

**Comment by shivamsharma01** [comment]
Yes, probably. I was able to setup the workspace with the help of @hhdqirui 's suggestions but now I am not able to login to the application. Don't know how to create a user.

**Comment by rom-me** [comment]
Can I also attempt working on this issue if it's still not resolved?

**Comment by t-cheepeng** [comment]
@rom-me Hi, please look for another issue to work on as there are previous users who wish to tackle this

**Comment by t-cheepeng** [comment]
@shivamsharma01 you can refer to https://github.com/TEAMMATES/teammates/blob/master/docs/development.md to see how to create users

**Comment by shivamsharma01** [comment]
In my understanding, there is a switch between the loading-spinner component (via loading-spinner directive) and the templateref using ngIf directive. There is no direct way to show the message while the loader is being displayed. I have added the logic to display 'Loading Records...' after 5 seconds but this is now not limited to records being loaded in the feedback page but every time the loader component is visible for more than 5 seconds on any screen. Kindly let me know if instead this should be limited to only a specific case and any suggestions to handle it.

![image](https://user-images.githubusercontent.com/21052557/156904100-a3f6844a-932c-4b91-a977-2ed5204c0975.png)

**Comment by hlee18lee46** [comment]
Can I take this issue?

**Comment by NicolasCwy** [comment]
@hlee18lee46 Go ahead!

**Comment by PrakritiShetty** [comment]
Can I work on this issue, or is it already assigned to someone?

**Comment by fsgmhoward** [comment]
@PrakritiShetty Someone else has taken it. Please find other issues to work on instead!

**Comment by sumit970** [comment]
Assign this issue to me

**Comment by samuelfangjw** [comment]
> Assign this issue to me

Our Development Workflow can be found [here](https://teammates.github.io/teammates/process.html).

**Comment by asn001013** [comment]
@sumit970 Hey! Are you still working on this? If not, I'd like to work on it

**Comment by MMilou24** [comment]
Hi, I would like to take this issue up

**Comment by domlimm** [comment]
@sumit970 Hello, are you still working on this issue? If not, we can have others to take this up.

@MMilou24 Hey there! Let's give @sumit970 some time to respond. If not, you may look at other issues [here](https://github.com/TEAMMATES/teammates/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) for first time contributors.

**Comment by domlimm** [comment]
@sumit970 Thanks for getting back! Much appreciated.

@MMilou24 Feel free to work on this issue.

**Comment by nielsly** [comment]
@MMilou24 and I are working on this together for a university course, therefore I have created the pull request.
